src/openforms/api/validators.py

```diff
@@ -10,21 +10,24 @@
 from .utils import get_model_serializer_instance
 
 
-class AllOrNoneRequiredFieldsValidator:
+class AllOrNoneTruthyFieldsValidator:
     """
-    Validate that the set of fields is present as soon as one field is provided.
+    Validate that either none or all fields have a 'non-empty' value.
```

Comment:
Maybe good to add a note about the bool behaviour too? The name of the validator already clarifies that this validates for truthy/falsy values, but maybe good to make it explicit also in the doc string?